File: Cuarto/Laboratorio/Talleres/RobocodeLabo/src/robotito/CombatStrategy.java
package robotito;

import robocode.*;

public interface CombatStrategy {
  void init(JuniorRobot r);

  void onTick(JuniorRobot r);

  void onScannedRobot(JuniorRobot r);

  void onHitWall(JuniorRobot r);

  void onHitRobot(JuniorRobot r);

  void onHitByBullet(JuniorRobot r);


}

File: Cuarto/Laboratorio/Talleres/RobocodeLabo/robots/RobotitoKotlin/CombatStrategy.kt
package robotitoKotlin

import robocode.JuniorRobot

abstract class CombatStrategy(protected val robot: JuniorRobot)  {
    val bulletHalfPower = 1.5
    val bulletMaxPower = 3.0

    abstract fun onTick()

    abstract fun onScannedRobot()

    abstract fun onHitWall()

    abstract fun onHitByBullet()

    fun onHitRobot(){
        robot.turnGunTo(robot.hitRobotAngle)
        robot.fire(this.bulletMaxPower)
    }

    open fun readyToAttack() : Boolean{
        return false
    }
}